### PR TITLE
Fixes a regex which prepares the prefix search in MongoDB

### PR DESCRIPTION
**Pull request by andyHa**

### CODE REVIEW

File: src/test/java/sirius/db/mongo/MongoFilterFactorySpec.groovy [hunk 1]
```diff
@@ -61,6 +61,34 @@ class MongoFilterFactorySpec extends BaseSpecification {
...
                 isPresent()
     }
+
+    def "prefix with leading number works"() {
+        when:
+        PrefixTestEntity test = new PrefixTestEntity()
+        test.setPrefix("1-test")
+        mango.update(test)
+        then:
+        mongo.find().
```
Comment (sabieber) [inline src/test/java/sirius/db/mongo/MongoFilterFactorySpec.groovy]: Formatting looks wild here

File: src/main/java/sirius/db/mixing/AccessPath.java [hunk 2]
```diff
@@ -31,27 +33,35 @@
     /**
      * Creates a new access path which appends the given prefix and accessor to the current access path.
      *
-     * @param prefix   the prefix to be appended to all field names to make them unique (composites might be
+     * @param name   the prefix to be appended to all field names to make them unique (composites might be
```
Comment (sabieber) [inline src/main/java/sirius/db/mixing/AccessPath.java]: Parameter name doesn't really match JavaDoc now

File: src/main/java/sirius/db/mixing/AccessPath.java [hunk 3]
```diff
@@ -31,27 +33,35 @@
...
         }
 
-        result.prefix = this.prefix + prefix;
+        result.prefix = qualify(name);
         result.accessor = this.accessor.andThen(accessor);
 
         return result;
     }
 
+    public String qualify(String name) {
```
Comment (sabieber) [inline src/main/java/sirius/db/mixing/AccessPath.java]: JavaDoc?

File: src/main/java/sirius/db/mixing/Property.java [hunk 4]
```diff
@@ -129,9 +139,10 @@ protected Property(@Nonnull EntityDescriptor descriptor, @Nonnull AccessPath acc
         this.propertyKey = field.getDeclaringClass().getSimpleName() + "." + field.getName();
         this.alternativePropertyKey = "Model." + field.getName();
         this.field.setAccessible(true);
-        this.name = accessPath.prefix() + field.getName();
+        this.name = accessPath.qualify(field.getName());
```
Comment (sabieber) [inline src/main/java/sirius/db/mixing/Property.java]: driveby

File: src/main/java/sirius/db/mixing/InvalidFieldException.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.db.mixing;
+
+/**
+ * Marker exception to make which field produced an error.
```
Comment (sabieber) [inline src/main/java/sirius/db/mixing/InvalidFieldException.java]: make -> mark

File: src/main/java/sirius/db/mixing/InvalidFieldException.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+ * <p>
+ * This is used as "cause" for <tt>HandledExceptions</tt> that are generated by a failed field check.
+ * This way, other frameworks can pick up this exception and extract the source field for an error message.
+ * <p>
+ * This is used by sirius-biz to highlight the field which caused an error when trying to saven an entity.
```
Comment (sabieber) [inline src/main/java/sirius/db/mixing/InvalidFieldException.java]: saven -> save

File: src/main/java/sirius/db/mixing/Property.java (hunk 4 above)
Comment (andyHa) [reply src/main/java/sirius/db/mixing/Property.java]: nope, this uses the method defined above -> refactoring

Comment (sabieber) [reply src/main/java/sirius/db/mixing/Property.java]: yeah buts in the getFullLabel commit, thats what i meant

**Pull request merged**